### PR TITLE
Reorganize compositor

### PR DESCRIPTION
**Pull request by TheNeikos**

### CODE REVIEW

**Comment by matthiasbeyer** [comment]
Looks good.

File: src/compositor/buffer/image_buffer.h [hunk 1]
```diff
@@ -28,7 +28,7 @@
 #ifndef __WS_BACKGROUND_SERVICE_H__
 #define __WS_BACKGROUND_SERVICE_H__
 
-#include "compositor/buffer.h"
+#include "buffer.h"
```
Comment (neithernut) [inline src/compositor/buffer/image_buffer.h]: Please make this absolute.

File: src/compositor/buffer/image_buffer.c [hunk 2]
```diff
@@ -8,7 +8,7 @@
 #include "util/arithmetical.h"
 #include "buffer.h"
 
-#include "background_surface.h"
+#include "image_buffer.h"
```
Comment (neithernut) [inline src/compositor/buffer/image_buffer.c]: Please make this absolute.

File: src/compositor/buffer/frame_buffer.h [hunk 3]
```diff
@@ -44,7 +44,7 @@
 #include <pthread.h>
 
 #include "objects/object.h"
-#include "compositor/background_surface.h"
+#include "image_buffer.h"
```
Comment (neithernut) [inline src/compositor/buffer/frame_buffer.h]: Please make this absolute.

File: src/compositor/buffer/frame_buffer.c [hunk 4]
```diff
@@ -38,8 +38,8 @@
 #include "logger/module.h"
 #include "objects/object.h"
 
-#include "compositor/background_surface.h"
-#include "compositor/frame_buffer.h"
+#include "image_buffer.h"
+#include "frame_buffer.h"
```
Comment (neithernut) [inline src/compositor/buffer/frame_buffer.c]: Please make this absolute.

File: src/compositor/buffer/buffer.c [hunk 5]
```diff
@@ -31,7 +31,7 @@
 #include "util/arithmetical.h"
 
 #include "compositor/internal_context.h"
-#include "compositor/buffer.h"
+#include "buffer.h"
```
Comment (neithernut) [inline src/compositor/buffer/buffer.c]: Please make this absolute.

File: src/compositor/module.c [hunk 6]
```diff
@@ -41,13 +41,13 @@
...
 #include "compositor/internal_context.h"
-#include "compositor/wayland_compositor.h"
-#include "compositor/wayland_shell.h"
+#include "compositor/wayland/wayland_compositor.h"
+#include "compositor/wayland/wayland_shell.h"
 #include "compositor/cursor.h"
-#include "background_surface.h"
+#include "buffer/image_buffer.h"
 #include "monitor_mode.h"
 #include "monitor.h"
```
Comment (neithernut) [inline src/compositor/module.c]: Please make those absolute, too.

**Comment by neithernut** [comment]
ACK.

**Pull request merged**